swbemservices.go

```diff
@@ -0,0 +1,287 @@
+// +build windows
+
+package wmi
+
+import (
+	"fmt"
+	"github.com/go-ole/go-ole"
+	"github.com/go-ole/go-ole/oleutil"
+	"reflect"
+	"runtime"
+)
+
+// SWbemServices is used to access wmi. See https://msdn.microsoft.com/en-us/library/aa393719(v=vs.85).aspx
+type SWbemServices struct {
+	//TODO: track namespace. Not sure if we can re connect to a different namespace using the same instance
+	cWMIClient            *Client //This could also be an embedded struct, but then we would need to branch on Client vs SWbemServices in the Query method
+	sWbemLocatorIUnknown  *ole.IUnknown
+	sWbemLocatorIDispatch *ole.IDispatch
+	queries               chan *queryRequest
+	closeError            chan error
+}
+
+type queryRequest struct {
+	query    string
+	dst      interface{}
+	args     interface{}
+	finished chan error
+}
+
+// InitializeSWbemServices will return a new SWbemServices object that can be used to query WMI
+func InitializeSWbemServices(c *Client, connectServerArgs ...interface{}) (*SWbemServices, error) {
+	fmt.Println("InitializeSWbemServices: Starting")
+	s := new(SWbemServices)
+	s.cWMIClient = c
+	s.queries = make(chan *queryRequest)
+	initError := make(chan error)
+	go s.process(initError)
+
+WaitForProcessSignal:
+	for {
+		select {
+		case err, ok := <-initError:
+			if ok {
+				return nil, err //Send error to caller
+			}
+			break WaitForProcessSignal
+		default:
```

Comment:
you also don't need the for/selects in these loops. Most of them look like a single channel read.